iceoryx_posh/source/popo/building_blocks/chunk_queue_popper.cpp

```diff
@@ -113,27 +113,41 @@ void ChunkQueuePopper::clear() noexcept
     } while (true);
 }
 
-/// @deprecated #25
-cxx::expected<ChunkQueueError>
-ChunkQueuePopper::attachSemaphore(const mepoo::SharedPointer<posix::Semaphore>& semaphore) noexcept
+bool ChunkQueuePopper::attachConditionVariableSignaler(ConditionVariableData* conditionVariableDataPtr) noexcept
 {
-    if (isSemaphoreAttached())
+    /// @todo Add lock guard here
```

Comment:
keep in mind, this doesn't need to be added only here and in detach. It has to be added everywhere it is accessed. I would suggest to put the two members into a struct and use the smart_lock for that struct

Reply:
Thanks for the hint. Using the `smart_lock` is a neat idea. Will implement it as part of the next step of #25 after #202 and #219 have been merged. Then we'll add the `ThreadSafePolicy` to both `ChunkQueuePopper`.

Reply:
well, is this much work?
If it doesn't blow up this PR, then I'd rather see it here. I fear we postpone to much to follow up PRs and might forget something and especially with concurrent code I'm rather save than sorry.

Reply:
@marthtz just starts working on introducing the locking policy. So this will not be a long lasting topic

Reply:
Is this a blocker for you? If you're fine with it, I'd prefer to do it in the follow-up. Upfront, promised ;-)